subsys/tracing/Kconfig

```diff
@@ -161,6 +161,15 @@ config TRACING_BACKEND_RAM
 	  be dumped to a file at runtime with a debugger.
 	  See gdb dump binary memory documentation for example.
 
+config TRACING_BACKEND_RAM_RING
+	bool "Enable RAM ring backend"
+	help
+	  Similar to TRACING_BACKEND_RAM, but older data is dropped
+	  in favor of newer data. GDB can be used to dump data,
+	  see gdb dump binary memory documentation for example.
```

Comment:
how about having this as a ram backend mode option outside the choice, something like

```
config TRACING_BACKEND_RAM_RING
	bool "RAM backend ring mode"
	depends on TRACING_BACKEND_RAM
	help
	  Execute RAM Tracing backend in ring mode.
```